pkg/apis/monitoring/v1/types.go

```diff
@@ -385,6 +385,21 @@ type PrometheusSpec struct {
 	// except if either value is zero, in which case the non-zero value will be
 	// used.  If both values are zero, no limit is enforced.
 	EnforcedTargetLimit *uint64 `json:"enforcedTargetLimit,omitempty"`
+	// Per-scrape limit on number of labels that will be accepted for a sample. If
+	// more than this number of labels are present post metric-relabeling, the
+	// entire scrape will be treated as failed. 0 means no limit.
+	// Only valid in Prometheus versions 2.27.0 and newer.
+	LabelLimit *uint64 `json:"labelLimit,omitempty"`
+	// Per-scrape limit on length of labels name that will be accepted for a sample.
+	// If a label name is longer than this number post metric-relabeling, the entire
+	// scrape will be treated as failed. 0 means no limit.
+	// Only valid in Prometheus versions 2.27.0 and newer.
+	LabelNameLengthLimit *uint64 `json:"label_name_length_limit,omitempty"`
+	// Per-scrape limit on length of labels value that will be accepted for a sample.
+	// If a label value is longer than this number post metric-relabeling, the
+	// entire scrape will be treated as failed. 0 means no limit.
+	// Only valid in Prometheus versions 2.27.0 and newer.
+	LabelValueLengthLimit *uint64 `json:"label_value_length_limit,omitempty"`
```

Comment:
In the same fashion as for the `EnforcedTargetLimit` field, we should prefix the label limits by `Enforced` since if defined at the Prometheus level, it will be enforced on all the ServiceMonitors/PodMonitors/Probes.

Reply:
I reviewed with TargetLimit, in [PrometheusSpec](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#prometheusspec) its named as `EnforcedTargetLimit` where as in [ServiceMonitorSpec](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#servicemonitorspec) and [PodMonitorSpec](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#podmonitorspec) its TargetLimit where user defined values taken